### PR TITLE
Add NewWithClient

### PR DESCRIPTION
**Pull request by nickatsegment**
So that we can use an http.Client with an stats.httpstats.transport

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -57,12 +57,22 @@ type Client struct {
 
 // New creates and returns a new Terraform Enterprise client
 func New(atlasToken string, baseURL string) *Client {
+	return NewWithClient(
+		atlasToken,
+		baseURL,
+		&http.Client{
```
Comment (systemizer) [inline client.go]: any reason not to use a vanilla http client (instead of setting timeout)? Just curious, but this looks great to me regardless!

Comment (dfuentes) [reply client.go]: The default timeout is no timeout.  Before I added the client timeout, I'd regularly see requests hange for 5+ minutes, then finally return 500 with empty bodies.

Comment (nickatsegment) [reply client.go]: ^ That sounds about right. But I was just maintaining the old behaviour.

Comment (systemizer) [reply client.go]: oh sorry i didn't realize this was inherited from the previous code!

**Pull request merged**